src/util/StringUtils.h

```diff
@@ -665,18 +263,19 @@ inline size_t findLiteralEnd(const std::string_view input,
 }  // namespace ad_utility
 
 // these overloads are missing in the STL
+// they can be constexpr once the compiler completely supports C++20
 inline std::string operator+(const std::string& a, std::string_view b) {
   std::string res;
-  res.resize(a.size() + b.size());
-  std::memcpy(res.data(), a.data(), a.size());
-  std::memcpy(res.data() + a.size(), b.data(), b.size());
+  res.reserve(a.size() + b.size());
+  res += a;
+  res += b;
```

Comment:
I was thinking about replacing all of the occurences of these functions, because there is a reason they don't exist.
But for now I agree with your refactorings.

Reply:
(My approach has less branches I assume, but has to zero the memory in the resize call.

Reply:
Also `std::memcpy` is not constexpr, starting with C++20, all of the above functions are (even though there isn't widespread compiler/library support yet).

Reply:
I agree, can you add a // TODO for the constexpr
(More aggressively: add a static_assert that string is not constexpr).
In a library we would use a macro CONSTEXPR_IF_STRING_IS_CONSTEXPR
but this leads too far here:)